### PR TITLE
Jt/persistent menu icons

### PR DESCRIPTION
**Pull request by Jtang-1**
## Technical Summary

Related to https://github.com/dimagi/commcare-hq/pull/34872 and https://github.com/dimagi/formplayer/pull/1594
[Jira Ticket
](https://dimagi.atlassian.net/browse/USH-4734)Adds the following to `persistentMenu` response to provide context to Web Apps on what icon to display.
- `imageUri` provides the uri for custom defined icons
- `navigationState` provides context to determine if the command is a form (`JUMP`), module (`NEXT`), or entity (no `navigationState` defined) 

### Safety story

tested locally. Change will be limited to apps that enable persistent menu.

### Automated test coverage

Automated tests exist that checks `persistentMenu` menu response is as expected. I added tests that the `navigationState` and `imageUri` are also correct in the response.

### QA Plan

TBD 

### Special deploy instructions

- [X] This PR can be deployed after merge with no further considerations.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations.

### Review

- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/beans/menus/PeristentCommand.java [hunk 1]
```diff
@@ -44,4 +58,20 @@ public ArrayList<PeristentCommand> getCommands() {
     public void addCommand(PeristentCommand command) {
         commands.add(command);
     }
+
+    public NavIconState getNavigationState() {
+        return navigationState;
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/beans/menus/PeristentCommand.java:63]: does this get used anywhere on Web Apps for menu screens ?

Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PeristentCommand.java:63]: Yes, it's used by web apps [here](https://github.com/dimagi/commcare-hq/blob/96da31d98ba42fadc12c8b2568652a7777a19209/corehq/apps/cloudcare/templates/cloudcare/partials/menu/row.html#L9-L13) to determine which default icon to use (the pencil for forms or folder for menus)

Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/menus/PeristentCommand.java:61]: Ahh that's super good to know, think it's worth to add a comment somewhere for this to describe what it does as the naming for this is quite obscure to know what it's used for

Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PeristentCommand.java:62]: [7aed2c5](https://github.com/dimagi/formplayer/pull/1603/commits/7aed2c59030a1920350acfd46f03537dfadac393)

**Review by shubham1g5 — COMMENTED** [review]
Noting that we should not change the `libs/commcare` branch head to a particular branch but keep it to `formplayer` branch head only as otherwise it causes unresolvable conflicts on staging.

File: src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java [hunk 2]
```diff
@@ -8,10 +12,13 @@
  * Model used to represent a persistent menu in FP HTTP API
  */
 @EqualsAndHashCode
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java:15]: It makes it so the response doesn't include keys that have null values.  I figured it's preferable to declutter the response by doing that.

File: src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java [hunk 3]
```diff
@@ -20,6 +27,14 @@ public class PersistentCommand {
...
 
+    public PersistentCommand(String index, String displayText,
+        String imageUri, NavIconState navigationState) {
+        this.index = index;
+        this.displayText = displayText;
+        this.imageUri = imageUri;
+        this.navigationState = navigationState;
+    }
+
     public PersistentCommand(String index, String displayText) {
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java:38]: hm i guess it's not necessary and in `addEntitySelection` we can instead call  `PersistentCommand(input, entityText, null, null)`. Is that preferable?

File: src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt [hunk 4]
```diff
@@ -23,13 +25,16 @@ class PersistentMenuHelper(val isPersistentMenuEnabled: Boolean) {
         }
     }
 
-    fun addMenusToPeristentMenu(menuScreen: MenuScreen) {
+    fun addMenusToPeristentMenu(menuScreen: MenuScreen, sessionWrapper: SessionWrapper) {
         if (isPersistentMenuEnabled) {
             val options = menuScreen.getOptions()
+        val mChoices = menuScreen.getMenuDisplayables()
```
[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt]: [4ef4143](https://github.com/dimagi/formplayer/pull/1603/commits/4ef4143b7d64f2102fd4a59e89bd765166c4bffc)

File: src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java:15]: yes that makes sense. I incorrectly read it to include nulls.

File: src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java:38]: Think they are both fine in general. Although think we would want to show an icon for entity selection as well and we should convey a new icon state to HQ for entity selections on the basic of which HQ can decide to show an icon corresponding to the case list selections.

File: src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt [hunk 5]
```diff
@@ -27,14 +27,15 @@ class PersistentMenuHelper(val isPersistentMenuEnabled: Boolean) {
...
     fun addMenusToPeristentMenu(menuScreen: MenuScreen, sessionWrapper: SessionWrapper) {
         if (isPersistentMenuEnabled) {
-            val options = menuScreen.getOptions()
-        val mChoices = menuScreen.getMenuDisplayables()
-            for (i in options.indices) {
-                val imageUri = mChoices[i].getImageURI();
-                val navigationState = CommandUtils.getIconState(mChoices[i], sessionWrapper);
+            val mChoices = menuScreen.getMenuDisplayables()
+            for (i in mChoices.indices) {
+                val d = mChoices[i];
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt:32]: nit: rename `d` -> `choice`

File: src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java]: The original intention was to have the absence of `navigationState` to denote case list selection. But I am open to defining a string value for it. Not sure what an appropriate string would be to follow the connotation of `NEXT` and `JUMP`... what do you think of `SELECT`?

[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java]: the simplification to one constructor is this commit [d1af633](https://github.com/dimagi/formplayer/pull/1603/commits/d1af633f0264d22c479dc79e0f7e215fa28bfd68)

[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java]: > what do you think of SELECT?

Think `ENTITY_SELECT` will be a clear one here.

[the remark this answers is not in the document]
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/beans/menus/PersistentCommand.java]: [0f2f639](https://github.com/dimagi/formplayer/pull/1603/commits/0f2f63947b0afb26976fe7fd800b7236103bd10f)

File: src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt (hunk 5 above)
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt]: [568266e](https://github.com/dimagi/formplayer/pull/1603/commits/568266e9415762e48eee5e2f6b421353a7c7c144)

**Review by shubham1g5 — DISMISSED** [review]

**Review by shubham1g5 — APPROVED** [review]

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**